File: 백준 16922번 로마숫자 만들기.cpp
#include<iostream>
#include<algorithm>
using namespace std;
bool checked[50*20+1];
int n,answer;
int main()
{
	ios::sync_with_stdio(false);
	cin.tie(NULL);
	cin>>n;
	for(int i=0; i<=n; i++)
	{
		for(int j=0; j<=n-i; j++)
		{
			for(int k=0; k<=n-i-j; k++)
			{
				int l=n-i-j-k;
				checked[i+5*j+10*k+50*l]=true;
			}
		}
	}
	for(int i=0; i<=50*20; i++)
	{
		if(checked[i])
		answer++;
	}
	cout<<answer;
}

File: Boj-Kotlin&Java/백준 16922번 로마숫자만들기.kt

private val checked=Array(50*20+1){false}
private var answer=0
fun main()
{
    val n=System.`in`.bufferedReader()
            .readLine().toInt()
    for(i in 0..n)
    {
        for(j in 0..n-i)
        {
            for(k in 0 ..n-i-j)
            {
                var l=n-i-j-k
                checked[i+5*j+10*k+50*l]=true
            }
        }
    }
    for(i in 0 ..50*20)
    {
        if(checked[i])
            answer++
    }
    println(answer)
}
